libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -1007,6 +1020,68 @@ static void _serializeUnsubscribe( const IotMqttSubscription_t * pSubscriptionLi
 
 /*-----------------------------------------------------------*/
 
+static IotMqttError_t _decodeSubackStatus( size_t statusCount,
+                                           const uint8_t * pStatusStart,
+                                           _mqttPacket_t * pSuback )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    uint8_t subscriptionStatus = 0;
+    size_t i = 0;
+
```

Comment:
validating arguments for being non null would be good.

Reply:
This is an internal function, so we can guarantee its arguments will not be NULL.